src/pl/plpython/plpy_main.c

```diff
@@ -451,6 +460,47 @@ PLy_pop_execution_context(void)
 
 	PLy_execution_contexts = context->next;
 
+	if (PLy_execution_contexts == NULL) {
+		// Clear pending interrupts when top level context exits
+		PLy_pending_interrupt = 0;
+	}
+
 	MemoryContextDelete(context->scratch_ctx);
```

Comment:
I'm wondering what will happend if we try to get the `scratch_ctx` from NULL. I'm pretty sure it will fail